db/migrate/20150611073002_create_assistantships.rb

```diff
@@ -0,0 +1,10 @@
+class CreateAssistantships < ActiveRecord::Migration
+  def change
+    create_table :assistantships do |t|
+      t.references :user, index: true
+      t.references :course, index: true
```

Comment:
I think there should be just one index of `(user_id, course_id)` and it should be unique.

Also, please add a database-level foreign key to both course and user with `:dependent => :delete` for both.

Please do all this for teacherships too.